src/rust/src/pkcs12.rs

```diff
@@ -386,6 +394,136 @@ enum CertificateOrPKCS12Certificate {
     PKCS12Certificate(pyo3::Py<PKCS12Certificate>),
 }
 
+fn serialize_bags<'p>(
+    py: pyo3::Python<'p>,
+    safebags: &[cryptography_x509::pkcs12::SafeBag<'_>],
+    encryption_details: &KeySerializationEncryption<'_>,
+) -> CryptographyResult<pyo3::Bound<'p, pyo3::types::PyBytes>> {
+    let mut auth_safe_contents = vec![];
+    let (
+        plain_safebag_contents,
+        shrouded_safebag_contents,
+        auth_safe_salt,
+        auth_safe_iv,
+        auth_safe_ciphertext,
+    );
+
+    if let Some(e) = &encryption_details.encryption_algorithm {
+        let mut shrouded_safebags = vec![];
+        let mut plain_safebags = vec![];
+        for safebag in safebags {
+            match safebag.bag_value.as_inner() {
+                cryptography_x509::pkcs12::BagValue::ShroudedKeyBag(_) => &mut shrouded_safebags,
+                _ => &mut plain_safebags,
+            }
+            .push(safebag);
```

Comment:
This would be easier to read as `=> shrouded_safebags.push(safebag)` (and the same for the other path)

Reply:
Completely agree. I've updated it. Not sure why I did it this way to begin with.